### ISSUE TITLE
getComputedRole returns null for option inside select due to Required Context Role mismatch

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3214]
## Summary

`getComputedRole()` returns `{ role: null, errorType: "INVALID_REQUIRED_CONTEXT_ROLE" }` for `<option>` elements inside `<select>`, because `matchesContextRole()` does not recognize `combobox` (the implicit role of `<select>`) as a valid parent role for the `option` role.

## Problem

The `option` ARIA role requires `requiredAccessibilityParentRole` of `listbox` or `group` (within listbox). However, `<select>` (without `multiple` or `size`) has an implicit role of `combobox`, not `listbox`. This causes `matchesContextRole()` to reject the parent context, and `getComputedRole()` resets the computed role to `null`.

### Impact

1. **`without` checks are skipped** — `checkingDisallowedProp` exits early when `role` is `null` (`if (!role) return;`), so element-specific restrictions like `aria-selected` should-not are never evaluated.
2. **False positive produced** — Instead of the correct "should not use" message, `checkingNoGlobalProp` fires with `"The \"aria-selected\" ARIA state is not global state"`.

### Reproduction

```html
<select><option selected aria-selected="true">A</option></select>
```

**Expected:** `"The \"aria-selected\" ARIA state should not use on the \"option\" element. As its state is already provided by the \"selected\" attribute"`

**Actual:** `"The \"aria-selected\" ARIA state is not global state"`

### Debug results

```
MLElement.matches(":is(select, select > option, datalist) > option") → true  ✓
getARIA() → { implicitRole: "option", properties: { without: [aria-selected] } }  ✓
getComputedRole() → { role: null, errorType: "INVALID_REQUIRED_CONTEXT_ROLE" }  ✗
```

## Root cause

`get-computed-role.ts` line 102:

```typescript
if (!matchesContextRole(computedRole.role.requiredAccessibilityParentRole, el, specs, version)) {
    return { el, role: null, errorType: 'INVALID_REQUIRED_CONTEXT_ROLE' };
}
```

The `option` role's `requiredAccessibilityParentRole` is `["group", "listbox"]`, but `<select>`'s implicit role is `combobox`. The HTML-ARIA spec explicitly allows `<option>` inside `<select>`, but the abstract ARIA role model does not list `combobox` as a valid context for `option`.

## Possible approaches

- Adjust `matchesContextRole()` to consider HTML-ARIA element mappings (not just abstract role requirements)
- Skip the Required Context Role check when the parent-child relationship is valid per HTML-ARIA spec
- Allow `combobox` as a valid context for `option` role

## Related

- Discovered while implementing #2465 (aria-valuenow restrictions)
- Affects spec.option.json `alt` field for `aria-selected` (added in #2465 but currently unreachable)
